src/core/src/factory.rs

```diff
@@ -425,24 +425,28 @@ pub trait Factory<R: Resources> {
     {
         self.view_texture_as_depth_stencil(tex, 0, None, texture::DepthStencilFlags::empty())
     }
-
-    fn create_texture_immutable_u8<T: format::TextureFormat>(&mut self, kind: texture::Kind, data: &[&[u8]])
+    
+    fn create_texture_immutable_u8<T: format::TextureFormat>(&mut self, kind: texture::Kind, mipmap: texture::Mipmap, data: &[&[u8]])
                                    -> Result<(handle::Texture<R, T::Surface>,
                                               handle::ShaderResourceView<R, T::View>),
                                              CombinedError>
     {
         let surface = <T::Surface as format::SurfaceTyped>::get_surface_type();
         let num_slices = kind.get_num_slices().unwrap_or(1) as usize;
         let num_faces = if kind.is_cube() {6} else {1};
+        let mut texture_level = (data.len() / (num_slices * num_faces)) as texture::Level;
+        if let texture::Mipmap::Allocated = mipmap {
+	        texture_level *= kind.get_num_levels();
```

Comment:
this is a rather non-trivial code for something that should be as trivial as
```rust
let levels = match mipmap {
   Allocated => {
               if data.len() != num_slices * num_faces {
                         return Err(...)
               }
               kind.get_num_levels()
   }
   Provided => {
               let num_slices = kind.get_num_slices().unwrap_or(1) as usize;
               let num_faces = if kind.is_cube() {6} else {1};
               (data.len() / (num_slices * num_faces)) as texture::Level
   }
};
```

Reply:
```rust
let levels = match mipmap {
		    texture::Mipmap::Allocated => if data.len() != num_slices * num_faces {
                return Err(CombinedError::Texture(texture::CreationError::Mipmap));
            } else {
                kind.get_num_levels()
            },
		    texture::Mipmap::Provided => (data.len() / (num_slices * num_faces)) as texture::Level
		};
        let desc = texture::Info {
            kind: kind,
            levels: levels,
            format: surface,
            bind: SHADER_RESOURCE,
            usage: Usage::Data,
        };
```